test/e2e/util_gatewayapi_test.go

```diff
@@ -741,3 +858,78 @@ func assertDNSRecord(t *testing.T, recordName types.NamespacedName) error {
 	})
 	return err
 }
+
+// assertVAP checks if the VAP of the given name exists, and returns an error if not.
+func assertVAP(t *testing.T, name string) error {
+	t.Helper()
+	vap := &admissionregistrationv1.ValidatingAdmissionPolicy{}
+	return wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 1*time.Minute, false, func(context context.Context) (bool, error) {
+		if err := kclient.Get(context, types.NamespacedName{Name: name}, vap); err != nil {
+			t.Logf("failed to get vap %q: %v, retrying...", name, err)
+			return false, nil
+		}
+		t.Logf("found vap %q", name)
+		return true, nil
+	})
+}
+
+// scaleDeployment scales the deployment with the given name to the specified number of replicas.
+func scaleDeployment(t *testing.T, namespace, name string, replicas int32) error {
+	t.Helper()
+
+	nsName := types.NamespacedName{Namespace: namespace, Name: name}
+	return wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 30*time.Second, false, func(context context.Context) (bool, error) {
+		depl := &appsv1.Deployment{}
+		if err := kclient.Get(context, nsName, depl); err != nil {
+			t.Logf("failed to get deployment %q: %v, retrying...", nsName, err)
+			return false, nil
+		}
+		if *depl.Spec.Replicas != replicas {
+			depl.Spec.Replicas = &replicas
+			if err := kclient.Update(context, depl); err != nil {
+				t.Logf("failed to update deployment %q: %v, retrying...", nsName, err)
+				return false, nil
+			}
+		}
+		t.Logf("scaled  deployment %q to %d replica(s)", nsName, replicas)
```

Comment:
You may want to check that the replicas actually deployed.

Reply:
The check for the actual number of available replicas is added.